### PR TITLE
updating search spyse cert task with API v4

### PR DESCRIPTION
**Pull request by AnasBensalah**
Updating search spyse cert task with recent Spyse API version v4 and adding new findings to the task:

image
image

### CODE REVIEW

File: lib/tasks/search_spyse_cert.rb [hunk 1]
```diff
@@ -1,83 +1,88 @@
...
-      _log_error "Unsupported entity type"
-    end
-
-  end #end run
+      def self.metadata
+        {
+          name: 'search_spyse_cert',
+          pretty_name: 'Search Spyse Cert',
+          authors: ['Anas Ben Salah'],
+          description: 'This task hits Spyse API for discovring domains registered with the same certificate',
```
Comment (shpendk) [inline lib/tasks/search_spyse_cert.rb:10]: typo in "discovering"

File: lib/issues/wildcard_certificate.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
+module Intrigue
+  module Issue
+    class WildcardCertificate < BaseIssue
+      def self.generate(instance_details={})
+        to_return = {
+          added: '2021-06-11',
+          name: 'wildcard_certificate',
+          pretty_name: 'Wildcard Certificate',
+          severity: 4,
```
Comment (shpendk) [inline lib/issues/wildcard_certificate.rb:9]: please change severity to 5

File: lib/tasks/search_spyse_cert.rb [hunk 3]
```diff
@@ -1,83 +1,88 @@
...
-      if result["parsed"]["names"]
-        result["parsed"]["names"].each do |domain|
-          _create_entity("DnsRecord", {"name" => domain })
-        end
-      end
+        json["data"]["items"].each do |result|
+          ## Create issues
+          # Create an issue if many domains founded registered with same Certificate
+          if result['parsed']['names']
+            _create_linked_issue('wildcard_certificate', {
```
Comment (shpendk) [inline lib/tasks/search_spyse_cert.rb:43]: How do we know this is a wildcard certificate?

**Review by shpendk — CHANGES_REQUESTED** [review]
See inline comments

**Comment by shpendk** [comment]
Changes as per our discussion:
The task itself should not create any issues. It should only collect new entities and create those. The issues `expired_cert` and `almost_expired_cert` are handled in enrichment. The wildcard certificate check should be moved to enrichment too, and verified on the certificate itself.

**Review by shpendk — CHANGES_REQUESTED** [review]
Running this against `amazon.com` returns the following error:
```
23:36:07 worker.1 | 2021-07-16T21:36:07.927Z pid=96853 tid=p49 WARN: NoMethodError: undefined method `[]' for nil:NilClass
23:36:07 worker.1 | 2021-07-16T21:36:07.927Z pid=96853 tid=p49 WARN: intrigue-core/lib/tasks/search_spyse_cert.rb:40:in `run'
23:36:07 worker.1 | intrigue-core/lib/tasks/base.rb:160:in `perform'

```

I also tested against an actual SslCertificate entity (which came from `uri_gather_ssl_certificate` task, and the error appears again. We need to do better error handling before we can merge this code.

**Comment by shpendk** [comment]
Have yet to find a time when the api key is not over quota to test this task.

**Review by AnasBensalah — COMMENTED** [review]
> Have yet to find a time when the api key is not over quota to test this task.

regex added and tested

**Comment by shpendk** [comment]
@AnasBensalah Can you please resolve the conflict and confirm this task is now working as expected? Thanks

**Comment by AnasBensalah** [comment]
> @AnasBensalah Can you please resolve the conflict and confirm this task is now working as expected? Thanks

@shpendk it should work fine but unfortunately, I can't test it since Spyse abandoned the free trial.

**Pull request merged**